### DISCUSSION TITLE
Support spline curves and G2 constraints

### DISCUSSION THREAD

**Discussion by Intuos5** [hlorus/CAD_Sketcher#96] [Feature Requests]
It would be great if the addon would support curves. It could be like the new Curve Pen tool  in Blender, the Flexibezier addon or the Fit point Spline in Fusion 360 (these work like the pen tool in illustrator). I would also like to see 
This does also imply the addition of a G2 Curvature constraint (and possibly G3). This all does add a lot of complexity though.

**Comment by hlorus** [comment]
I'm not exactly sure what you mean by the term "curve" here. Anyway the types of entities that can be implemented are limited to the ones that the solver supports. That is the ones that are already supported by the addon plus the bezier entity. 

There is some info in the solvespace documentation: https://solvespace.com/ref.pl

**Reply by Intuos5** [reply]
'Curve' is used in other software to denote any splines, whether they are fit point (through point) or CV splines: 
So these things: 
![afbeelding](https://user-images.githubusercontent.com/63149655/166469919-02524555-0749-42c3-86fa-eda688bbf291.png)
So forget about the curve terminology, because in Blender, a curve is a container for line and curve objects. And that's not what I am referring to. I am merely refering to the spline type of lines, the drawing methods (as in Illustrator/ Inscape/ Fusion 360) and the end result: curved surfaces/ meshes that are not derivatives of a circle, but higher degree 'curves'. Hope this makes sense.

**Reply by Intuos5** [reply]
@hlorus the documentation includes Cubic Bezier Splines, so I am hoping to see a Spline drawing tool and and G2 constraints (curvature constraints) between splines.

**Reply by hlorus** [reply]
> @hlorus the documentation includes Cubic Bezier Splines, so I am hoping to see a Spline drawing tool and and G2 constraints (curvature constraints) between splines. 

At some point for sure, it's just not a high priority for me right now but if someone wants to work on this i'm here to assist.

**Reply by aarch-angel** [reply]
> > @hlorus the documentation includes Cubic Bezier Splines, so I am hoping to see a Spline drawing tool and and G2 constraints (curvature constraints) between splines.
> 
> At some point for sure, it's just not a high priority for me right now but if someone wants to work on this i'm here to assist.

I just started using CAD Sketcher, thanks for the great add-on! I'd be willing to try working on this. Do you have any specific ideas / thoughts on implementing this?

**Reply by hlorus** [reply]
@Cosmic-Goat that would be great! I've posted some infos in discord's contribute channel: https://discord.com/channels/882152824249667594/967316874121588777/969291169441935441

The file structure has changed a bit sinc then, so now class_defines.py is plit into individual files in the model folder.

Let me know if you have any specific questions.

**Reply by aarch-angel** [reply]
> @Cosmic-Goat that would be great! I've posted some infos in discord's contribute channel: https://discord.com/channels/882152824249667594/967316874121588777/969291169441935441
> 
> The file structure has changed a bit sinc then, so now class_defines.py is plit into individual files in the model folder.
> 
> Let me know if you have any specific questions.

Great, thank you! I've taken a brief look at the code, I'll let you know if I'm able to make any progress or have any questions! :)
